src/debuggability.js

```diff
@@ -141,6 +141,69 @@ Promise.hasLongStackTraces = function () {
     return config.longStackTraces && longStackTracesIsSupported();
 };
 
+var supportedEvents = ["created", "chained", "fulfilled",
+    "rejected", "resolved", "cancelled"];
+
+Promise.on = function (eventName, hookFunction, simpleEventApi) {
```

Comment:
You can add the promises as additional properties of the event object, that is fully compatible